stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/usecase/MarkReadImpl.kt

```diff
@@ -19,12 +20,19 @@ public interface MarkRead {
 }
 
 internal class MarkReadImpl(private val domainImpl: ChatDomainImpl) : MarkRead {
-    override operator fun invoke(cid: String): Call<Boolean> {
-        validateCid(cid)
+    override operator fun invoke(cid: String): Call<Boolean> =
+        domainImpl.channel(validateCid(cid)).let { channelController ->
+            CoroutineCall(domainImpl.scope) {
+                channelController.markRead().let { markedRead ->
```

Comment:
Is there a reason to introduce all this nesting here with `let` instead of just assigning the results of the calls to regular variables?

Reply:
Seems like it's just a style preference.